urdf2webots/parserURDF.py

```diff
@@ -594,13 +594,16 @@ def getVisual(link, node, path, outputDirectory):
             link.visual.append(visual)
         elif hasElement(geometryElement, 'mesh'):
             meshfile = geometryElement.getElementsByTagName('mesh')[0].getAttribute('filename')
-            if not os.path.isabs(meshfile):
-                # Use the path relative to the output file
-                meshfile = os.path.normpath(os.path.relpath(os.path.join(path, meshfile), outputDirectory))
             # hack for gazebo mesh database
             if meshfile.count('package'):
                 idx0 = meshfile.find('package://')
                 meshfile = meshfile[idx0 + len('package://'):]
+            elif meshfile.count('file'):
+                idx0 = meshfile.find('file://')
+                meshfile = meshfile[idx0 + len('file://'):]
+            if not os.path.isabs(meshfile):
+                # Use the path relative to the output file
+                meshfile = os.path.normpath(os.path.relpath(os.path.join(path, meshfile), outputDirectory))
```

Comment:
That would seem clearer and cleaner to me:
```suggestion
            if meshfile.startswith('package://'):
                idx0 = meshfile.find('package://')
                meshfile = meshfile[idx0 + len('package://'):]  # package relative path
            elif meshfile.startswith('file:///'):
                idx0 = meshfile.find('file:///')
                meshfile = meshfile[idx0 + len('file://'):]  # we keep the last slash for absolute path
            elif not os.path.isabs(meshfile):
                # Use the path relative to the output file
                meshfile = os.path.normpath(os.path.relpath(os.path.join(path, meshfile), outputDirectory))
```